### PR TITLE
Add documentation for dot notation in arrays, objects, and embedded documents

### LINKED ISSUE #1310: Add documentation on embedded documents and dot notation support for arrays
### What should be done?

Add a section on document queries, embedded documents, array queries, and Dot notation support for arrays. 

* [x] add section for dot notations in the document section
* [x] retrieve documents arrays using dot notation
* [x] update documents arrays using dot notation
* [x] query an embedded document using dot notation
* [x] update an embedded document using dot notation

### PR DESCRIPTION
**Pull request by Fashander**
# Description

Closes #1310.

## Readiness checklist

* [ ] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: website/docs/Basic_operations/read.md [hunk 1]
```diff
@@ -85,6 +85,98 @@ Here is a list of the most commonly used operators.
...
+```
+
+### Retrieve documents containing an array element using dot notation
+
+To retrieve all documents containing a specific value in an array index, use dot notation to reference its position in the `employees` collection.
+The following command retrieves all documents containing `"blender"` in the third element of an array:
+
+```sh
+db.employees.find({"catalog.2": "blender"})
+```
```
Comment (chilagrow) [inline website/docs/Basic_operations/read.md:178]: Let's add an response here too!

File: website/docs/understanding_ferretdb.md [hunk 2]
```diff
@@ -56,6 +56,69 @@ For complex documents, you can nest objects (subdocuments) inside a document.
...
+When using dot notations, the field name of the array and the specified element must be enclosed in quotation marks.
+:::
+
+For example, let's take the following array field in a document:
+
+```sh
+animals: ["dog", "cat", "fish", "fox"]
+```
+
+To reference the fourth element in the array, use the dot notation `" animals.4"`.
```
Comment (chilagrow) [inline website/docs/understanding_ferretdb.md:81]:
```suggestion
To reference the fourth element in the array, use the dot notation `"animals.3"`.
```
Could you double check if index starts start 0 or 1? previous example made me think index starts at 0.

File: website/docs/understanding_ferretdb.md [hunk 3]
```diff
@@ -56,6 +56,69 @@ For complex documents, you can nest objects (subdocuments) inside a document.
...
+
+Here are more examples of dot notations on arrays:
+
+* [Query an array](https://docs.ferretdb.io/Basic_operations/read/#retrieve-documents-containing-a-value-in-an-array)
+* Update an array
```
Comment (chilagrow) [inline website/docs/understanding_ferretdb.md:86]: We want a link here right?

File: website/docs/understanding_ferretdb.md [hunk 4]
```diff
@@ -56,6 +56,69 @@ For complex documents, you can nest objects (subdocuments) inside a document.
...
+}
+```
+
+To reference the `city` field in the embedded document, use the dot notation `" contact.address.city"`.
+
+For dot notation examples on embedded documents, see here:
+
+* Query an embedded document
+* Update an embedded document
+* Query an array of embedded documents
```
Comment (chilagrow) [inline website/docs/understanding_ferretdb.md:120]: Let's also add link or examples here :hugs:

**Review by chilagrow — CHANGES_REQUESTED** [review]

File: website/docs/understanding_ferretdb.md (hunk 3 above)
Comment (Fashander) [reply website/docs/understanding_ferretdb.md:86]: The sections for the links are not ready yet. I'll be updating them once they are ready.

File: website/docs/understanding_ferretdb.md (hunk 4 above)
Comment (Fashander) [reply website/docs/understanding_ferretdb.md:120]: Same as above

**Review by AlekSi — COMMENTED** [review]
Also, "Conform PR" check fails

File: website/docs/understanding_ferretdb.md (hunk 3 above)
Comment (chilagrow) [reply website/docs/understanding_ferretdb.md:86]: Would it be on a separate PR?

Comment (Fashander) [reply website/docs/understanding_ferretdb.md:86]: Yes. Also makes it easier to review in chunks than the whole thing at once.

**Review by chilagrow — DISMISSED** [review]

File: website/src/pages/index.js [hunk 5]
```diff
@@ -30,7 +30,7 @@ export default function Home() {
   const {siteConfig} = useDocusaurusContext();
   return (
     <Layout
-      title={`Hello from ${siteConfig.title}`}
+      title={`${siteConfig.title}`}
       description="Description will go into a meta tag in <head />">
```
Comment (AlekSi) [inline website/src/pages/index.js:34]: Maybe update that too while we are on it?

File: website/docs/understanding_ferretdb.md [hunk 6]
```diff
@@ -56,6 +56,69 @@ For complex documents, you can nest objects (subdocuments) inside a document.
 
 In the example above, the `name` field is a subdocument embedded into a document.
 
+## Dot Notation
+
+Dot notations `(.)` are used to specify the elements in an array, the objects in a document, or the values in an embedded document.
```
Comment (AlekSi) [inline website/docs/understanding_ferretdb.md:61]: What is the difference between objects in document and values in the embedded document?

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: website/docs/Basic_operations/read.md [hunk 7]
```diff
@@ -85,10 +85,117 @@ Here is a list of the most commonly used operators.
...
+
+To retrieve all documents containing a specific array element and value (`catalog: "printer"`), run the following command:
+
+```js
+db.employees.find({catalog: "printer"})
+```
+
+The response displays all the retrieved documents:
+
+```sh
```
Comment (rumyantseva) [inline website/docs/Basic_operations/read.md:150]:
```suggestion
```js
```

**Review by chilagrow — DISMISSED** [review]

File: website/docs/understanding_ferretdb.md [hunk 8]
```diff
@@ -56,6 +56,69 @@ For complex documents, you can nest objects (subdocuments) inside a document.
...
+      address:{
+         city: "Kent",
+         state: "Ohio"
+      },
+      phone: "432-124-1234"
+   }
+}
+```
+
+To reference the `city` field in the embedded document, use the dot notation `" contact.address.city"`.
```
Comment (rumyantseva) [inline website/docs/understanding_ferretdb.md:114]:
```suggestion
To reference the `city` field in the embedded document, use the dot notation `"contact.address.city"`.
```

File: website/docs/Basic_operations/read.md [hunk 9]
```diff
@@ -85,10 +85,117 @@ Here is a list of the most commonly used operators.
 
 `$eq`: select records that are equal to a specific value
 
+### Retrieve documents containing a value in an array
+
+Insert the following documents in an `employees` collection using this command:
+
+```js
+db.employees.insertMany([
```
Comment (rumyantseva) [inline website/docs/Basic_operations/read.md:93]: I'm not sure if in this example formatting is consistent, e.g. whitespace seem to be missing to me.
I've see we have this issue https://github.com/FerretDB/FerretDB/issues/1389, so hopefully, we can automate formatting.

**Review by rumyantseva — DISMISSED** [review]
Looks good to me overall, left a couple of minor comments.

**Review by chilagrow — DISMISSED** [review]

**Review by rumyantseva — DISMISSED** [review]
LGTM

**Review by noisersup — DISMISSED** [review]

**Review by w84thesun — APPROVED** [review]

**Comment by Fashander** [comment]
> Waiting for the promised update

**Review by AlekSi — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

**Pull request merged**